gateway/gateway.go

```diff
@@ -616,7 +617,12 @@ func (g *gateway) updateDeviceInfo(device *node.Node, d *deviceInfo) error {
 	device.AppSKey = appsKey
 	device.Addr = savedAddr
 	device.NwkSKey = nwksKey
-	device.FCntDown = d.FCntDown
+
+	// if we don't have an FCntDown in the device file, set it to a max number so we can tell.
+	device.FCntDown = math.MaxUint32
+	if d.FCntDown != nil {
+		device.FCntDown = *d.FCntDown
+	}
```

Comment:
we prob don't have to checkboth the NwkSKey and FCntDown since the two of them were done at the same time. doesn't hurt to keep this tho